pkg/state/release.go

```diff
@@ -107,13 +108,18 @@ func (r ReleaseSpec) ExecuteTemplateExpressions(renderer *tmpl.FileRenderer) (*R
 					return nil, fmt.Errorf("failed executing template expressions in release \"%s\".values[%d] = \"%v\": %v", r.Name, i, string(serialized), err)
 				}
 
-				var deserialized map[interface{}]interface{}
+				var deserialized map[string]interface{}
```

Comment:
yaml.v2 deserializes nested yaml dict to `map[interface{}]interface{}` while yaml.v3 and goccy/go-yaml serializes it to `map[string]interface{}`.
My original design was to ensure any nested dict is deserialized to the latter, by calling `maputil.CastKeysToStrings`